### ISSUE TITLE
Setting permalink in 11tydata.json works for every file except the index.njk

### ISSUE THREAD

**Issue by simonsarris** [11ty/eleventy#1627]
**Describe the bug**
Setting `"permalink": "intro/{{ page.fileSlug }}.html"` in a `11tydata.json` ought to allow an arbitrary `blah.njk` file to be built as `blah.html` instead of `/blah/index.html`

**To Reproduce**
Suppose we have:

```
src/intro
src/intro/index.njk
src/intro/blah.njk
src/intro/intro.11tydata.json
```

And in the intro.11tydata.json:

```
{
  "permalink": "intro/{{ page.fileSlug }}.html"
}
```

**Expected behavior**

The output should contain two files:

intro/index.html
intro/blah.html

but instead of **index**.html it is **intro**.html

This seems like an unexpected edge case of the permalink. Is it a bug, or intentional?

**Environment:**: Windows and 0.11.1

**Comment by simonsarris** [comment]
Note that a workaround to force index.njk -> index.html when using this permalink JSON is to add front-matter to only index.njk:

```
---
permalink: "intro/index.html"
---
```

**Comment by btrem** [comment]
What is the `slug` value in intro.njk?

**Comment by pdehaan** [comment]
@simonsarris What about something like:

```json
{
  "permalink": "{{ page.filePathStem }}.html"
}
```

```sh
$ tree src www

src
├── _includes
│   └── layouts
│       └── base.njk
└── intro
    ├── about.njk
    ├── blah.njk
    ├── index.njk
    ├── intro.11tydata.json
    └── nested
        ├── about.njk
        └── contact.njk
www
└── intro
    ├── about.html
    ├── blah.html
    ├── index.html
    └── nested
        ├── about.html
        └── contact.html

6 directories, 12 files
```

**Comment by pdehaan** [comment]
@btrem Here's what I saw locally with a demo repo when dumping the `page` object in the ./src/intro/index.njk file:

```js
{
  date: '2021-02-11T00:35:27.011Z',
  inputPath: './src/intro/index.njk',
  fileSlug: 'intro',
  filePathStem: '/intro/index',
  url: '/intro/',
  outputPath: 'www/intro/index.html'
}
```

Note that my `fileSlug` is "intro" (which is the parent directory's name) and not "index" (which is the current template's name). But that only happens on the index.* pages, if I remember correctly.

Per https://www.11ty.dev/docs/data-eleventy-supplied/#page-variable-contents,

> ```
>  // For permalinks: inputPath filename minus template file extension (New in v0.3.4)
>  fileSlug: "myFile",
>
>  // For permalinks: inputPath minus template file extension (New in v0.9.0)
>  filePathStem: "/current/page/myFile",
> ```

**Comment by simonsarris** [comment]
Yes thank you @pdehaan, it does seem like a more acceptable workaround to write
`"permalink": "{{ page.filePathStem }}.html"`

Instead of
`"permalink": "intro/{{ page.fileSlug }}.html"`

In the config. That's certainly a solution enough for my use case.

But it still seems like a bug that the `fileSlug` works for all file names except the one that so-happens to be named `index.njk`. I would perhaps the correct file slug in the page object would be the empty string, instead of `'intro'`, but I cannot know what's intentional.

**Comment by danburzo** [comment]
Just wanted to chip in that I also stumbled upon this issue recently. The `page.fileSlug` behavior in regards to index files — that is to report the parent folder name rather than the empty string — makes it more difficult to remap permalinks for an entire folder.

```
src/
  folder/
    a.md
    b.md
    index.md
```

The `folder.11tydata.json` replaces `folder/` with a different prefix, but generates `prefix/folder/` instead of `prefix/` for the index file. 

```json
{ 
  "permalink": "prefix/{{page.fileSlug}}/"
}
```

For now, I am going to override `permalink` in the index file as [suggested above](#issuecomment-774136714):

```md
---
permalink: 'prefix/'
---
```
